lib/ansible/plugins/action/debug.py

```diff
@@ -61,8 +59,10 @@ def run(self, tmp=None, task_vars=None):
                             raise AnsibleUndefinedVariable
```

Comment:
Since we can now display the exception message, this AnsibleUndefinedVariable would be better with an explanation message. something like:

 ``` python
raise AnsibleUndefinedVariable('The result of templating  was expected to be text type, but instead was a "%s" type. The  template input: %s" % (self._task.args['var'], type(results)))
```

Reply:
and maybe a blurb indicating that error comes from withing the debug action itself,

'While "debug" task was trying to display the value of "%s" the result of templating  was expected to be text type, but instead was a "%s" type.' % (type(results), self._task.args['var'],)

Reply:
At a higher level... why is this being triggered?  (Does it get triggered?)  jinja2 returns strings always.  If templar is returning non-strings, it's because we take a string from jinja2 and turn it into something else in our code.  In either of those cases, debug should still be able to output a sensible value so we shouldn't be raising an exception for it.

Reply:
not something i changed in this case, leaving as is for this PR